esmvalcore/_config/_diagnostics.py

```diff
@@ -77,6 +77,13 @@ class TagsManager(dict):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        # Set a default author for testing purposes
```

Comment:
Adding code just to be able to run tests is a bit awkward. Could you create a config-references.yml file with the required information for running your tests instead?

Reply:
I agree it is a bit awkward, but creating a new `config-references.yml` means we are committing to creating a mini-repo for testing. The way the diagnostics and tags are initiated (with the dependency on `esmvaltool` and fall-back to nothing) make it very difficult to test this properly.

I tried the monkeypatching route, but this became complex very quickly, because the tags import needs to be overwritten in several places, and it caused the tests to fail in other places that monkeypatched in tags.

I eventually figured it out with monkeypatching, but there are two tests that test Julia / R support through a subprocess, and these bypass any variables that we can set. In absence of an `esmvaltool` import, they will initialize `config-references.yml` (TAGS) as an empty dict. Yet, authors are a required keyword. So writing the html file requires the authors to be resolvable, and will thus crash the program.

Adding a new `config-references.yml` also does not crate a clean path to fix the two tests. The issue is that there is no way to specify the location of `config-references.yml` (or diagnostics path for that matter). I will happily create an issue for it and work on it separately, but to me this seemed out of scope for this PR. 

I hope you understand why adding a default to the TAGS makes sense not only for the tests, but also to have a dummy author for new contributors that they can use while they are developing a recipe (#28).

Reply:
Hey @stefsmeets this sounds reasonable and I think it's a good compromise. Let's see if we can make time to address #28 separately.

Reply:
To summarize the offline discussion we had about this: running ESMValCore without a config-references.yml file is a feature that we would like to keep. The current behaviour is to not replace any tags if config-references.yml is missing. If config-references.yml is present, the tool will try to replace all tags and raise an exception if a tag that is used in the recipe or diagnostic script is not defined in config-references.yml.

Reply:
I made it so that it keeps the old behaviour. 👍

However, it creates a very fragile foundation to test from.

1. User defined an author tag that is in `esmvaltool` -> works both with and without `esmvaltool`
2. User defines author tag *not* in `esmvaltool` -> works only when `esmvaltool` is *not* installed, crashes when `esmvaltool` *is* installed

We should have a way to make the define of the diagnostics path i.e. `config-references.yml` so that the outcome of a test no longer depends on whether `esmvaltool` is installed or not.